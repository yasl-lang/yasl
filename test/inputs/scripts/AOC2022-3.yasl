
# https://adventofcode.com/2022/day/3
const set = collections.set

const rucksacks = `
vJrwpWtwJgWrhcsFMMfFFhFp
jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL
PmmdzqPrVvPwwTWBwg
wMqvLMZHhHMvwLHjbvcjnnSBnvTQFn
ttgJtRGJQctTZtZT
CrZsJsPPZsGzwwsLwLmpwMDw
`->trim()->split()

const fn ord(char) {
    const a = char->tolower()->tobyte() - 'a'->tobyte() + 1
    return char == char->toupper() ? a + 26 : a
}

echo ord('c'), ord('A')


let tmp = 0
for rucksack in rucksacks {
    let a, let b = rucksack->tolist(len rucksack // 2)->spread()
    a, b = set(a->spread()), set(b->spread())
    for char in a & b {
        tmp += ord(char)
    }
}

echo tmp
