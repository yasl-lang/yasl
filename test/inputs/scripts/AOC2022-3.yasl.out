3, 27
157
